# 1. Buat array dasar:

Buat array 1D berisi angka 1–10 menggunakan np.arange(), lalu buat array yang sama menggunakan np.linspace() sehingga hasilnya identik. Bandingkan hasilnya.

In [1]:
import numpy as np

In [6]:
range = np.arange(1, 11)
print(range)

linspace = np.linspace(1, 10, 10)
print(linspace)

[ 1  2  3  4  5  6  7  8  9 10]
[ 1.  2.  3.  4.  5.  6.  7.  8.  9. 10.]


# 2. Eksperimen shape

Buat array 2D berukuran 3x4 berisi angka 1–12 (gunakan np.arange(1,13).reshape(3,4) — boleh cari tahu dulu fungsi .reshape()). Cetak .shape, .ndim, dan .size-nya, lalu jelaskan dengan kata-katamu sendiri kenapa hasilnya seperti itu.

In [ ]:
np2d = np.arange(1, 13).reshape(3, 4)
print(np2d)
print(np2d.ndim)
print(np2d.shape)
print(np2d.size)

[[ 1  2  3  4]
 [ 5  6  7  8]
 [ 9 10 11 12]]
2
(3, 4)
12


* ndim (number of dimension): 2 > dimensi 1 kolom, dimensi 2 baris
* shape: (3, 4) > reshape(3, 4) > 3 baris 4 kolom
* size: 12 > 3x4 = 12 > jumlah seluruh elemen dalam array.

# 3. Perbandingan list vs array

Buat list Python [1,2,3] dan NumPy array [1,2,3]. Coba kalikan keduanya dengan angka 3 (* 3), amati hasilnya, dan tuliskan penjelasan kenapa hasilnya berbeda.

In [10]:
list = [1, 2, 3]
arr = np.array([1,2, 3])

print(list * 3)
print(arr * 3)

[1, 2, 3, 1, 2, 3, 1, 2, 3]
[3 6 9]


* list hanya menggandakan element bukan dikalikan perelement
* array mengalikan perelemen

# 4. Simulasi kasus dunia kerja: Kenaikan Gaji

Buat array gaji = np.array([5000000, 7500000, 6200000, 9000000]). Hitung gaji baru setelah kenaikan 8% menggunakan operasi array (tanpa loop). Cetak hasilnya.

In [12]:
gaji = np.array([5000000, 7500000, 6200000, 9000000])
gajiup = gaji * 0.08
gajifinal = gaji + gajiup

print(gajifinal)

[5400000. 8100000. 6696000. 9720000.]


# 5. Dtype dan memori

Buat array np.array([1,2,3]), cek .dtype-nya. Lalu ubah ke int8 dengan .astype(np.int8). Cek lagi dtype-nya. Coba jelaskan: kalau kamu punya array berisi angka 1–1000, apakah aman diubah ke int8? Kenapa? (hint: cari tahu batas maksimum int8)

In [14]:
arr5 = np.array([1, 2, 3])
print(arr5.dtype)

arr5 = np.array([1, 2, 3]).astype(np.int8)
print(arr5.dtype)

int64
int8


int8 adalah integer yang menggunakan 8 bit untuk menyimpan satu angka.

In [15]:
print(np.iinfo(np.int8))

Machine parameters for int8
---------------------------------------------------------------
min = -128
max = 127
---------------------------------------------------------------



int8 maksimum = 127
1000 > 127

* diluar range int8
* nilai yang diluar range mengalami overflow / wrapping > hasilnya tidak mempresetasika anga aslinya

| Dtype   | Ukuran |                  Range integer |
| ------- | -----: | -----------------------------: |
| `int8`  | 1 byte |                     -128 → 127 |
| `int16` | 2 byte |               -32.768 → 32.767 |
| `int32` | 4 byte | -2.147.483.648 → 2.147.483.647 |
| `int64` | 8 byte |                   sangat besar |
